codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -0,0 +1,129 @@
+  succeeded,
+}
+
+/// Codesign and notarize all files within a [RemoteArchive].
+class FileCodesignVisitor extends FileVisitor {
+  FileCodesignVisitor({
+    required this.commitHash,
+    required this.codesignCertName,
+    required this.codesignUserName,
+    required this.appSpecificPassword,
+    required this.codesignAppstoreId,
+    required this.codesignTeamId,
+    required this.codesignFilepaths,
+    this.production = false,
+  });
+
+  /// Temp [Directory] to download/extract files to.
+  ///
+  /// This file will be deleted if [validateAll] completes successfully.
+  Directory? tempDir;
+  FileSystem? fileSystem;
+  Stdio? stdio;
+  ProcessManager? processManager;
+
+  final String commitHash;
+  final String codesignCertName;
+  final String codesignUserName;
+  final String appSpecificPassword;
+  final String codesignAppstoreId;
+  final String codesignTeamId;
+  final bool production;
+
+  // TODO(xilaizhang): add back utitlity in later splits
+  Set<String> fileWithEntitlements = <String>{};
+  Set<String> fileWithoutEntitlements = <String>{};
+  Set<String> fileConsumed = <String>{};
+  List<String> codesignFilepaths;
+
+  late final File entitlementsFile;
+  late final Directory remoteDownloadsDir;
+  late final Directory codesignedZipsDir;
+
+  int _remoteDownloadIndex = 0;
+  int get remoteDownloadIndex => _remoteDownloadIndex++;
+
+  static const String _entitlementsFileContents = '''
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+    <dict>
+        <key>com.apple.security.cs.allow-jit</key>
+        <true/>
+        <key>com.apple.security.cs.allow-unsigned-executable-memory</key>
+        <true/>
+        <key>com.apple.security.cs.allow-dyld-environment-variables</key>
+        <true/>
+        <key>com.apple.security.network.client</key>
+        <true/>
+        <key>com.apple.security.network.server</key>
+        <true/>
+        <key>com.apple.security.cs.disable-library-validation</key>
+        <true/>
+    </dict>
+</plist>
+''';
+
+  void initialize() {
+    fileSystem ??= LocalFileSystem();
+    tempDir ??= fileSystem!.systemTempDirectory.createTempSync('conductor_codesign');
+    stdio ??= VerboseStdio(
+      stdout: io.stdout,
+      stderr: io.stderr,
+      stdin: io.stdin,
+    );
+    processManager ??= LocalProcessManager();
+    entitlementsFile = tempDir!.childFile('Entitlements.plist')..writeAsStringSync(_entitlementsFileContents);
+    remoteDownloadsDir = tempDir!.childDirectory('downloads')..createSync();
+    codesignedZipsDir = tempDir!.childDirectory('codesigned_zips')..createSync();
+  }
+
+  Future<void> validateAll() async {
+    initialize();
+
+    try {
+      await Future.value(null);
+      stdio!.printStatus('Codesigned all binaries in ${tempDir!.path}');
+    } finally {
+      if (production) {
+        await tempDir?.delete(recursive: true);
```

Comment:
Why is cleaning the temp dir only for production?

Reply:
umm my understanding is for non-production mode (debugging/ testing), we might want to inspect the artifacts we downloaded (its structure), and inspect which files caused the error ( if an error occurred, such as notarization, or codesign, or entitlements, or upload, or zip failed). If we keep the tempDir, which contains our files and progress, we can better track down the errors, and see which folders are downloaded so far and the filesystem structure at the time of error.

Reply:
We're likely going to have an integration test that runs dry run. Since this will run on infra bots, the code should clean up after itself, unless there was some failure in the verification process.

Reply:
Thanks for explaining! Changed to cleanup for dryrun and production.